llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1991,13 +2015,12 @@ getDependenceDistanceStrideAndSize(
     return MemoryDepChecker::Dependence::Unknown;
   }
 
-  uint64_t TypeByteSize = DL.getTypeAllocSize(ATy);
   bool HasSameSize =
       DL.getTypeStoreSizeInBits(ATy) == DL.getTypeStoreSizeInBits(BTy);
```

Comment:
There is a difference now with the handling of scalable types. `getTypeStoreSizeInBits`/`getTypeAllocSize` return the minimum size. Now `TypeByteSize` is 0 if `ATy` is a scalable type (and happens to have the same minimum size as the `TypeStoreSize` of `BTy`), but not if the other way around with `BTy`.

I think it's safer to pass `0` for such cases (minimum size is probably not safe for a scalable type, why wasn't the maximum size used?), so maybe do the same for `BTy`?

Reply:
Thanks, I've updated the code to only query `DL.getTypeAllocSize` if non of the accessed types is scalable.

At the moment, those sizes will only be used by the newly added code; `getPtrStride` will return 0 for scalable types, so we will for now always return an unknown dependence due to the stride checks above.